### PR TITLE
Admin UI for confirming users

### PR DESCRIPTION
**Pull request by ashfurrow**
This PR exposes accounts' confirms statuses in the admin list page, and adds a button to the edit page to confirm them. I'm still pretty new at Rails, so please let me know if there are more idiomatic approaches to some of my decisions.

screen shot 2017-04-20 at 8 05 35 pm
screen shot 2017-04-20 at 8 05 41 pm

Fixes https://github.com/tootsuite/mastodon/issues/1297

### CODE REVIEW

File: app/controllers/admin/confirms_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Admin
+  class ConfirmsController < BaseController
```
Comment (mjankowski) [inline app/controllers/admin/confirms_controller.rb]: This is a little nitpicky, but you asked for rails feedback...

I would probably put this action either at:

- `Admin::UsersController#update` - with a route like /admin/users/123 - and make sure the only thing which can be updated is the confirmed_at; or
- `Admin::Confirmations#create` with a route like /admin/users/123/confirmation - and have that find that user and confirm it.

File: app/controllers/admin/confirms_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    before_action :set_account
+
+    def create
+      @account.user.update(confirmed_at: Time.now.utc)
```
Comment (mjankowski) [inline app/controllers/admin/confirms_controller.rb]: I believe devise provides a convenience method like `user.confirm` which does essentially what you are doing, with slightly more logic around it; so I might use that.

File: app/controllers/admin/confirms_controller.rb (hunk 1 above)
Comment (ashfurrow) [inline app/controllers/admin/confirms_controller.rb]: Okay, I _think_ I've got the second option done. Let me know what you think 👍

File: config/routes.rb [hunk 3]
```diff
@@ -87,6 +87,7 @@
       resource :reset, only: [:create]
       resource :silence, only: [:create, :destroy]
       resource :suspension, only: [:create, :destroy]
+      post '/confirmation', to:'confirmations#create'
```
Comment (ykzts) [inline config/routes.rb]: `resource :confirmation, to: [:create]` ?

Comment (ashfurrow) [inline config/routes.rb]: Cool, thanks 👍

**Review by Gargron — APPROVED** [review]

File: app/views/admin/accounts/show.html.haml [hunk 4]
```diff
@@ -77,6 +77,9 @@
   - else
     = link_to t('admin.accounts.silence'), admin_account_silence_path(@account.id), method: :post, class: 'button'
 
+  - unless @account.user.confirmed?
```
Comment (nightpool) [inline app/views/admin/accounts/show.html.haml]: This will break for non-local accounts, leading to a 500 error.

Comment (ashfurrow) [inline app/views/admin/accounts/show.html.haml]: Good catch, I'll submit a fix. Thanks!

**Pull request merged**